packages/mdc-typography/_mixins.scss

```diff
@@ -35,6 +35,10 @@
   font-weight: #{map-get($style-props, font-weight)};
   letter-spacing: map-get($style-props, letter-spacing);
   line-height: map-get($style-props, line-height);
+
+  @if $style == "button" {
```

Comment:
I am not sure if `text-decoration: none` will work for every other typography, for example links, I don't think we are opinionated that links cannot use `mdc-typography`, so I am a bit hesitate to do that.

So it seems like what you want to achieve here is just make button less special, how about use `text-decoration:  inherit` and `text-transform: inherit` then?